pkg/sql/opt/xform/coster.go

```diff
@@ -134,13 +133,269 @@ const (
+	"st_asewkt":                   100 * cpuCostFactor,
+	"st_asgeojson":                100 * cpuCostFactor,
+	"st_asgml":                    100 * cpuCostFactor,
+	"st_ashexewkb":                100 * cpuCostFactor,
+	"st_askml":                    100 * cpuCostFactor,
+	"st_aslatlontext":             100 * cpuCostFactor,
+	"st_assvg":                    100 * cpuCostFactor,
+	"st_astext":                   100 * cpuCostFactor,
+	"st_astwkb":                   1000 * cpuCostFactor,
+	"st_asx3d":                    100 * cpuCostFactor,
+	"st_azimuth":                  100 * cpuCostFactor,
+	"st_bdmpolyfromtext":          100 * cpuCostFactor,
+	"st_bdpolyfromtext":           100 * cpuCostFactor,
+	"st_boundary":                 1000 * cpuCostFactor,
+	"st_boundingdiagonal":         100 * cpuCostFactor,
+	"st_box2dfromgeohash":         1000 * cpuCostFactor,
+	"st_buffer":                   100 * cpuCostFactor,
+	"st_buildarea":                10000 * cpuCostFactor,
+	"st_centroid":                 100 * cpuCostFactor,
+	"st_chaikinsmoothing":         10000 * cpuCostFactor,
+	"st_cleangeometry":            10000 * cpuCostFactor,
+	"st_clipbybox2d":              10000 * cpuCostFactor,
+	"st_closestpoint":             1000 * cpuCostFactor,
+	"st_closestpointofapproach":   10000 * cpuCostFactor,
+	"st_clusterdbscan":            10000 * cpuCostFactor,
+	"st_clusterintersecting":      10000 * cpuCostFactor,
+	"st_clusterkmeans":            10000 * cpuCostFactor,
+	"st_clusterwithin":            10000 * cpuCostFactor,
+	"st_collectionextract":        100 * cpuCostFactor,
+	"st_collectionhomogenize":     100 * cpuCostFactor,
+	"st_concavehull":              10000 * cpuCostFactor,
+	"st_contains":                 10000 * cpuCostFactor,
+	"st_containsproperly":         10000 * cpuCostFactor,
+	"st_convexhull":               10000 * cpuCostFactor,
+	"st_coorddim":                 100 * cpuCostFactor,
+	"st_coveredby":                100 * cpuCostFactor,
+	"st_covers":                   100 * cpuCostFactor,
+	"st_cpawithin":                10000 * cpuCostFactor,
+	"st_createtopogeo":            100 * cpuCostFactor,
+	"st_crosses":                  10000 * cpuCostFactor,
+	"st_curvetoline":              10000 * cpuCostFactor,
+	"st_delaunaytriangles":        10000 * cpuCostFactor,
+	"st_dfullywithin":             10000 * cpuCostFactor,
+	"st_difference":               10000 * cpuCostFactor,
+	"st_dimension":                100 * cpuCostFactor,
+	"st_disjoint":                 10000 * cpuCostFactor,
+	"st_distance":                 100 * cpuCostFactor,
+	"st_distancecpa":              10000 * cpuCostFactor,
+	"st_distancesphere":           100 * cpuCostFactor,
+	"st_distancespheroid":         1000 * cpuCostFactor,
+	"st_dump":                     1000 * cpuCostFactor,
+	"st_dumppoints":               100 * cpuCostFactor,
+	"st_dumprings":                1000 * cpuCostFactor,
+	"st_dwithin":                  100 * cpuCostFactor,
```

Comment:
hmm, i wonder if we should take the maximum instead of the minimum.

Reply:
I also wondered that! I think for now minimum is fine (since it's at least better than what we had before), and we can assume this will get fixed if someone ever moves these costs to the overloads.